internal/controller/drplacementcontrol.go

```diff
@@ -2214,7 +2216,14 @@ func updateDRPCProgression(
 	drpc *rmn.DRPlacementControl, nextProgression rmn.ProgressionStatus, log logr.Logger,
 ) bool {
 	if drpc.Status.Progression != nextProgression {
-		log.Info(fmt.Sprintf("Progression: Current '%s'. Next '%s'",
+		// caller of this function is always d.setProgression()
+		// caller of d.setProgression() makes the progression decision.
+		// Use ancestorLevel=2 to get the caller of the caller.
+		// nolint: mnd
+		decisionFunction := getCallerFunction(2)
```

Comment:
so for the initial deployment, isn't this going to end up calling `processPlacement`. I didn't check the others, but maybe getCallerFunction(1) is enough.

Reply:
You're correct. I thought that call is from `setProgression()`